monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,234 @@
+            self.template_configs = template_configs
+        self.meta_data_filename = meta_data_filename
+        self.cfg = ConfigParser(globals=False)  # TODO: define root folder (variable)?
+        if self.template_configs is not None:
+            self.load_templates(self.template_configs, meta_data_filename, parser_args)
+
+        self.scripts_path = scripts_path
+        self.data_stats_files = None
+        self.data_list_file = None
+        self.output_path = None
+        self.name = None
+        self.best_metric = None
+
+    def load_templates(self, config_files, metadata_file=None, parser_args=None):
+        parser_args = parser_args or {}
+        self.cfg.read_config(config_files, **parser_args)
+        if metadata_file is not None:
+            self.cfg.read_meta(metadata_file)
+
+    def set_data_stats(self, data_stats_files):
+        self.data_stats_files = data_stats_files
+
+    def set_data_source(self, data_list_file):
+        self.data_list_file = data_list_file
+
+    def fill_template_config(self, data_stats_filename, **kwargs):
+        pass
+
+    def export_to_disk(self, output_path, algo_name, **kwargs):
+        self.fill_template_config(self.data_stats_files, **kwargs)
+        write_path = os.path.join(output_path, algo_name)
+        self.cfg["bundle_root"] = write_path
+        os.makedirs(write_path, exist_ok=True)
+        # handling scripts files
+        output_scripts_path = os.path.join(write_path, "scripts")
+        if os.path.exists(output_scripts_path):
+            shutil.rmtree(output_scripts_path)
+        if self.scripts_path is not None and os.path.exists(self.scripts_path):
+            shutil.copytree(self.scripts_path, output_scripts_path)
+        # handling config files
+        output_config_path = os.path.join(write_path, "configs")
+        if os.path.exists(output_config_path):
+            shutil.rmtree(output_config_path)
+        os.makedirs(output_config_path, exist_ok=True)
+        output_config_file = os.path.join(output_config_path, "algo_config.yaml")
+        ConfigParser.export_config_file(self.cfg.config, output_config_file, fmt="yaml", default_flow_style=None)
+        with open(output_config_file, "r+") as f:
+            lines = f.readlines()
+            f.seek(0)
+            f.write(f"# Generated automatically by `{__name__}`\n")
+            f.write("# For more information please visit: https://docs.monai.io/\n\n")
+            for item in ensure_tuple(self.template_configs):
+                f.write(f"# source file: {item}\n")
+            f.write("\n\n")
+            f.writelines(lines)
+        logger.info(write_path)
+        self.output_path = write_path
+
+    def train(self, params=None):
+        train_py = os.path.join(self.output_path, "scripts", "train.py")
+        config_yaml = os.path.join(self.output_path, "configs", "algo_config.yaml")
+        base_cmd = f"{train_py} run --config_file {config_yaml} "
+
+        if torch.cuda.device_count() > 1:
```

Comment:
Is it possible to manually set whether to run multi-gpu and how many GPUs to use? For example, users may have DGX and don't want to use all the GPUs for this task.

Thanks.